apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/Rates/index.jsx

```diff
@@ -70,13 +76,31 @@ export default function ProviderRate({
   );
 
   const setDexRate = useCallback(
-    provider => {
+    (provider) => {
       setDexSelected(provider);
       updateSelection(provider);
     },
     [updateSelection],
   );
 
+  useEffect(() => {
+    if (filter.includes(FILTER.decentralised)) {
+      setDexRate(DEX_PROVIDERS[0]);
+    } else {
+      if (filter.includes(FILTER.float)) {
+        const selectedRate = rates.find((rate) => rate.tradeMethod === FILTER.float);
+        setRate(selectedRate || rates[0]);
```

Comment:
dumb question but is it possible that rates[0] is undefined? is setRate ready to receive an undefined value?

Reply:
Good catch! 
I have add a empty object if the value is undefined and I have unify the empty status (I had missing delete a empty state legacy)